### ISSUE TITLE
Integrate with Build Farm & Grayskull

### ISSUE THREAD

**Issue by wolfv** [TheSnakePit/quetz#15]
It could be cool to integrate quetz with grayskull and a "build farm" (the build farm could e.g. be github repositories as in the case of conda-forge).
Grayskull project has already explored a webservice to generate recipes from pypi packages.

If we had that as a plugin we could automatically generate new recipes as needed from the frontend of quetz and add them to the build queue, and then they would appear on the package server.

**Comment by tom--pollard** [comment]
Has any thought/effort been put into looking at integrating with [`conda-smithy`](https://github.com/conda-forge/conda-smithy) in terms of  the "build farm" goal?

Definitely interested to see if Quetz could be used as an endpoint for existing conda-forge feedstocks!

**Comment by wolfv** [comment]
Nope, not really more thoughts as of yet! But thanks for bringing it up.

I would be interested to show some indicators for the packages in quetz -- for example a button that links to the feedstock, and an indicator that shows how many PR's against the feedstock are currently open. That should be quite easily doable.

**Comment by SylvainCorlay** [comment]
One way to add a notion of feedstock/recipe already is to add the notion of a group that can push to a bunch of packages, which are the outputs of a recipe.

The recipe owners could request changing permissions when adding / removing outputs.

**Comment by tom--pollard** [comment]
For what it's worth, I've been looking at taking existing conda-forge feedstocks, and vendoring the [setup package](https://github.com/conda-forge/conda-forge-ci-setup-feedstock) to push to a local instance of a Quetz server via quetz-client. Supporting for having a vendored setup-package is somewhat supported already in conda-smithy, although it will need to be fleshed out a bitter better to provide an alternative for non Anaconda Cloud / Binstar based uploads.

**Comment by wolfv** [comment]
@tom--pollard we are doing similar experiments – in fact, we just forked conda-smithy to optionally add quetz-client to it. 
The quetz packages are also on conda-forge now :)

**Comment by tom--pollard** [comment]
@wolfv The smithy feedstock I forked is [here](https://github.com/tom--pollard/conda-smithy-feedstock/commits/master). Interested to hear/see what the plans for Quetz in this regard are, do you plan to use this issue for the work? :) If there's going to be a repo for the smithy client work / setup package that would be great!

**Comment by wolfv** [comment]
hey @tom--pollard do you guys want to have a video chat with us at some point?

Our current plans are to set up the full loop: using conda-smithy to "generate" a feedstock that uses recipes generated by `vinca` that work with `boa` to create the ROS packages and upload them to a quetz instance we have set up using `quetz-client`.

Just to give some reference: ROS recipes can be quite easily generated using the vinca toolkit github.com/ros-forge/vinca
We'll be using this for testing for now.

**Comment by tom--pollard** [comment]
I think at some point that'd make sense @wolfv :) I'll raise it.

It's really cool to see the work towards ROS in conda, I remember not having fun with all the dependencies whilst at University but things seem to have come a long way since those days! If `vinca` generated recipes are just being used as an example usecase to seed feedstocks (in the sameway as using grayskull/skeleton/forking existing feedstocks etc) then having the work needed on `conda-smithy` & the `ci-setup-package` will be generic. Using `boa` & `mamba` inplace of conda/conda-build also brings benefits.

Will the fork to conda-smithy & its setup package be hosted under TheSnakePit, or just PR's to the respective upstreams? Either way it would be cool to collaborate.
